sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -127,19 +128,24 @@ object JavaTypeInference {
 
         // TODO: we should only collect properties that have getter and setter. However, some tests
         // pass in scala case class as java bean class which doesn't have getter and setter.
-        val properties = getJavaBeanReadableProperties(other)
-        val fields = properties.map { property =>
-          val returnType = typeToken.method(property.getReadMethod).getReturnType
-          val (dataType, nullable) = inferDataType(returnType, seenTypeSet + other)
-          new StructField(property.getName, dataType, nullable)
+        if (typeToken.getRawType.isEnum) {
```

Comment:
Probably, but int is cheaper. However, if we use string, there will be a possibility for meaningful queries. Also, there are strange cases with complex enums with different fields, even with complex types. I'd say string with constant name is enough, thou